agent/__main__.py

```diff
@@ -57,6 +72,9 @@ def main():
     elif args.action == 'daemon':
         logger.info("start in daemon mode...")
         run_daemon(dev=args.dev)
+    elif args.action == 'patch':
+        patch(args.patch_name, dev=args.dev)
+        run(ping=True, dev=args.dev)
```

Comment:
It should in theory be safe to run `service ssh restart` after, as it has a check-config first:

```
  restart)
	check_for_upstart 1
	check_privsep_dir
	check_config
	log_daemon_msg "Restarting OpenBSD Secure Shell server" "sshd" || true
	start-stop-daemon --stop --quiet --oknodo --retry 30 --pidfile /var/run/sshd.pid
	check_for_no_start log_end_msg
	check_dev_null log_end_msg
	if start-stop-daemon --start --quiet --oknodo --pidfile /var/run/sshd.pid --exec /usr/sbin/sshd -- $SSHD_OPTS; then
	    log_end_msg 0 || true
	else
	    log_end_msg 1 || true
	fi
	;;
```

Reply:
Sorry forgot about that

Reply:
Alternatively, we could check it manually (which is what `check_config` does above):

`/usr/sbin/sshd $SSHD_OPTS -t`

And then send a SIGHUP to it. 

That way it becomes distro agnostic.

Reply:
SIGHUP won’t restart it, it will reload. This is what “service sshd reload” does. Also I think (need to double-check) reload will fail if config is invalid.

Reply:
Yes, `service ssh reload` will do the same, however, the reason for sending SIGHUP vs using `service` is that we don't need to worry about distro specific things (like if it's called "ssh" or "sshd")

Reply:
Are we sure that it is in /usr/sbin in all distros?

Reply:
Nope. That said, I doesn't seem like the `-t` argument requires an absolute path:
```
$ sudo sshd -t
$ echo $?
0
```

(Launching sshd will however require an absolute path.)